cluster_experiments/power_analysis.py

```diff
@@ -524,3 +524,43 @@ def check_inputs(self):
         self.check_target_col()
         self.check_treatment()
         self.check_clusters()
+
+
+class PowerAnalysisWithPreExperimentData(PowerAnalysis):
+    """
+    Same as PowerAnalysis, but adding pre-experiment param during split and perturbation. The PowerAnalysis class
+    accepts a pre_experiment param, however this is only used for cuped purposes (where we add a column to df). This
+    child class is used for cases where the pre experiment df is also available when the class is instantiated
+    """
+
+    def _split(self, df: pd.DataFrame) -> pd.DataFrame:
+        treatment_df = self.splitter.assign_treatment_df(df)
+        self.log_nulls(treatment_df)
+
+        treatment_df = treatment_df.query(
+            f"{self.treatment_col}.notnull()", engine="python"
+        ).query(
+            f"{self.treatment_col}.isin(['{self.treatment}', '{self.control}'])",
+            engine="python",
+        )
+
+        return treatment_df
+
+    def _perturbate(
+        self, treatment_df: pd.DataFrame, average_effect: Optional[float]
+    ) -> pd.DataFrame:
+
+        df, pre_experiment_df = self.analysis._split_pre_experiment_df(treatment_df)
+
+        perturbed_df = self.perturbator.perturbate(df, average_effect=average_effect)
+
+        return pd.concat([perturbed_df, pre_experiment_df])
+
+    def _split_and_perturbate(
```

Comment:
I'd consider moving this in the parent class, remove the _split method in this class, and just override the _perturbate